### ISSUE TITLE
[BUG] Regex route with alias, with using generate

### ISSUE THREAD

**Issue by petrck** [krasimir/navigo#125]
Hi, I found bug, when I create regex route with alias (I does mean "as"), after this, I call generate method, for this route, and I got this:

`result.replace is not a function`

problem is here:

```
if (route.name === name) {
     result = route.route;
         for (key in data) {
                 result = result.replace(':' + key, data[key]);
         }
    }
return result;
```

when route.route is regular expression, cannot have function replace, as string have.

Is possible fix it please? Thank you. #

**Comment by krasimir** [comment]
Good catch. I'll take a look soon.

**Comment by krasimir** [comment]
I just released `4.7.3` which fixes the failing. However, the generated string is probably not what you want. And that's because there's no a simple way to fulfill the regex. When the route is a string is easy because it is a simple find and replace. While with the regex we need an analyze to understand where the capture groups are. I feel that this is out of the scope of this library.

**Comment by krasimir** [comment]
Just a note that in version 8.0.0 the named routes and routes as regexp are depricated.
